texttospeech/snippets/long_audio_quickstart.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# All Rights Reserved.
+
+from google.cloud import texttospeech
+import time
+import argparse
+
+def synthesize_long_audio(text, language_code, voice_name, parent, output_gcs_uri):
+  """
+  Synthesizes long input, writing the resulting audio to `output_gcs_uri`.
+  
+  Example usage: synthesize_long_audio('Some input text to synthesize', 'en-US', 'en-US-Standard-A', 
+  'projects/{PROJECT_NUMBER}/locations/{LOCATION}', 'gs://{BUCKET_NAME}/{OUTPUT_FILE_NAME}.wav')
+  
+  """
+
+  client = texttospeech.TextToSpeechLongAudioSynthesizeClient()
+
+  input = texttospeech.SynthesisInput()
+
+  input.text = input_text
+
+  audio_config = texttospeech.AudioConfig(audio_encoding=texttospeech.AudioEncoding.LINEAR16)
+
+  voice = texttospeech.VoiceSelectionParams(language_code=language_code, name=voice_name)
+
+  request = texttospeech.SynthesizeLongAudioRequest(parent=parent, input=input, audio_config=audio_config, voice=voice, output_gcs_uri=output_gcs_uri)
+
+  operation = client.synthesize_long_audio(request=request)
+
+  attempts = 0
+  while (!operation.done and attempts < 30):
+    print("Operation is still not done. Sleeping for 5 seconds and then trying again...")
+    time.sleep(5)
+    attempts += 1
+  if attempts == 30:
+    print("It's taking a while for your operation to complete. If the input was very large, this may be okay. If not, it likely means there was a failure.")
+    return False
+  print("Your operation is complete; check your GCS bucket to find your audio file!"
+  return True
```

Comment:
I'm not super familiar with this service, but this feels like a little bit of a workaround when I am guessing based on other services that we have a neater way to await the operation finishing. Could this sample instead utilize the [async client](https://github.com/googleapis/python-texttospeech/blob/main/google/cloud/texttospeech_v1/services/text_to_speech_long_audio_synthesize/async_client.py#L234)? If that's going to be a separate sample and this is explicitly to utilize the sync client, please utilize the [format described in the client library docstring](https://github.com/googleapis/python-texttospeech/blob/474744180b2a148c513ece4ba15fa66206c70d8d/google/cloud/texttospeech_v1/services/text_to_speech_long_audio_synthesize/client.py#L498-L506) rather than a while loop


Also tagging @telpirion who I believe has more TTS experience than I do

Reply:
hmm okay, it is async. Another way to do this could be to just return the operation itself, in this sample, and not wait for it to be done. Then, in the test, we can implement the logic of waiting for it to be done, or just assert that the operation was created as expected. I think that might make sense, because we don't need to tell the developer what to do once they have the operation, just want to show them how to create the operation using our API. What do you think?

Reply:
I would remove this. You can set a long timeout and await for results instead of polling like this.

Here's an example from a different sample & service:
https://github.com/googleapis/python-videointelligence/blob/main/samples/shotchange/shotchange.py#L50

Reply:
@dandhlee re your comment - this is the example Eric gave

Reply:
Where did we land on this topic? I see that this sample still has the polling logic.

Reply:
tried to implement with a timeout, but then I realized that is not supported for this service, since operation.result() is unsupported. Given that, I think the best way to do this is to poll, so I re-implemented that logic. We may support operation.result() in the future, in which case we could update the docs to use timeout.

Reply:
actually, we discussed and are going to implement this. I'll get back to you once this is in prod (may be like a week or so). Thanks!